### DISCUSSION TITLE
Next/Image not working with cloudinary

### DISCUSSION THREAD

**Discussion by uinstinct** [vercel/next.js#18538] [Help] [answered]
My images are loading fine with the normal _`img` tag_ but not with `next/image`.
I am uploading my images to _**Cloudinary**_.

In my **`next.config.js`**,
```js
module.exports = {
    images: {
        domains: ["res.cloudinary.com"],
        loader: "cloudinary",
    },
}
```

In one of my **components**:
```js
sourceItem.imgSrc = "https://res.cloudinary.com/gamersinstinct7/image/upload/v1604062891/portfolio/Character%20Card%20Builder/CCB_create_1_hzcd34.png";
sourceItem.title = "My cool image";
return(
<div className="imageContainer">
   <NextImage src={`${sourceItem.imgSrc}`} alt={`${sourceItem.title}`}
                                loading="lazy" unsized quality="100" />
</div>
);
```

**But, this does not work.**

I would be very grateful if someone can help me with this problem. If possible, please provide your configuration for _cloudinary_ for `next/image`.

**Comment by uinstinct** [comment, marked as answer]
Okay, this helped me:
this works for **ABSOLUTE urls**.
```js
module.exports = {
    images: {
        domains: ["res.cloudinary.com"],
    },
}
```


If you are using a **`loader`** like "_cloudinary_ you will have to use the **relative path**. In this case, the `_next/static` url will be prepended in each src of the image.


But, if you are going to use an **absolute url,** you should use the `domain` in `images` of `next.config.js`

**Comment by gayratv** [comment]
@uinstinct ,
Thank you for solution!
It's work for me too.
But I can't understand what you write about using  loader and the relative path.

I'm write:
```
module.exports = {
  images: {
    loader: "cloudinary",
  },
};
```

And try to insert to src this image:
```
{
  "url": "http://res.cloudinary.com/dljazkzna/image/upload/v1604677935/img/1053.jpg",
  "etag": "a4209fb1ea1b830680c23c8bb8bb408d",
  "tags": ["img" ],
  "type": "upload",
  "width": 780,
  "format": "jpg",
  "height": 780,
  "version": 1604677935,
  "asset_id": "c0b0349f5d04709034b2e920d071174a",
  "public_id": "img/1053",
  "signature": "e8020c83a74710be4e9cc489bf52cf2bdc77854d",
  "created_at": "2020-11-06T15:52:15Z",
  "secure_url": "https://res.cloudinary.com/dljazkzna/image/upload/v1604677935/img/1053.jpg",
  "version_id": "4cab9f6e19f50be8322acecaed0594e3",
  "placeholder": false,
  "resource_type": "image",
  "original_filename": "1053"
}
dljazkzna - is the cloud name
```

What's the realtive path I must pass to src props ?
I'm try src="/dljazkzna/image/upload/v1604677935/img/1053.jpg" - it's not work

**Comment by nandorojo** [comment]
Don't we lose next's size optimizations if we don't use relative URLs? I can't figure this out. Is there a codesandbox using cloudinary?

**Comment by dotlouis** [comment]
Just chiming in because I had the same issue.
I could not find how to use relative URL for Cloudinary and benefit from Next's loader automatic optimization.

Here's what I did to get it working:
```JS
// next.config.js
  images: {
    loader: 'cloudinary',
    path: 'https://res.cloudinary.com/<my-workspace>/image/upload/'
  }
```

_Notice there is no `/v1263537652` in the path. It's because it's an optional path parameter linked to cloudinary image versioning. You can omit it to get the latest media version_

Then on the image component:
```JS
<Image src={myCloudinaryResponse.public_id} width="150" height="150" />
// public_id should look like this: "myWorkspace/123456778"
```

It worked for me. You get the best image size, resolution, and format.

**Reply by KeithRussel** [reply]
What if i have images from public folder and I want to map the images from cloudinary also?

```
`import Image from 'next/image

const blog = ({posts}) => {
return (
<div className={styles.gridColumns}>
        {posts.posts.map((post, index) => {
          return (
            <div key={post.slug}>
              <Image
                src={post.feature_image}
                alt={post.title}
                width={500}
                height={500}
              />
              <Link href='/post/[slug]' as={`/post/${post.slug}`}>
                <a>{post.title}</a>
              </Link>
            </div>
          );
        })}
      </div>`
```

`// next.config.js module.exports = { images: { domains: ['res.cloudinary.com', 'localhost'], }, }`

**Comment by KeithRussel** [comment]
Same issue here. In my case my cloudinary is not working also.
Here's my code
`
// next.config.js
module.exports = {
  images: {
    domains: ['res.cloudinary.com', 'localhost'],
  },
}
`
And I want to show my features image dynamically from ghost cms and cloudinary for image storage
```

// blog.js
import Image from 'next/image

const blog = ({posts}) => {
return (
<div className={styles.gridColumns}>
        {posts.posts.map((post, index) => {
          return (
            <div key={post.slug}>
              <Image
                src={post.feature_image}
                alt={post.title}
                width={500}
                height={500}
              />
              <Link href='/post/[slug]' as={`/post/${post.slug}`}>
                <a>{post.title}</a>
              </Link>
            </div>
          );
        })}
      </div>
)}
```

**Reply by ervandra** [reply]
I'm recently going through this `cloudinary` and `next/image` optimization.

From what i learn, you need to handle manually your cloudinary optimization, for your case, your ghost CMS should already returning cloudinary image urls.

Then you're just serving the cloudinary using `next/image`

Otherwise, you might need to use `loader` which then you handle the cloudinary upload, get the result, then return to the `next/image` through your `loader`.

Also make sure your returning image urls from ghost CMS is already absolute url.
